Docker python tests broken at HEAD

Running the following test at HEAD (following the format of the Docker CI tests) fails:
```
docker-compose build
docker run --rm qsim-py-tests:latest
```
Error logs:
```
make: Entering directory '/qsim'
make -C pybind_interface/ pybind
make[1]: Entering directory '/qsim/pybind_interface'
g++ pybind_main.cpp -o ../qsimcirq/qsim`python3-config --extension-suffix` -O3 -march=native -fopenmp -Wall -shared -std=c++17 -fPIC `python3 -m pybind11 --includes`
make[1]: Leaving directory '/qsim/pybind_interface'
for exe in qsimcirq_tests/qsimcirq_test.py; do if ! python3 -m pytest $exe; then exit 1; fi; done
============================= test session starts ==============================
platform linux -- Python 3.7.3, pytest-3.10.1, py-1.7.0, pluggy-0.8.0
rootdir: /qsim, inifile:
collected 0 items / 1 errors

==================================== ERRORS ====================================
_______________ ERROR collecting qsimcirq_tests/qsimcirq_test.py _______________
ImportError while importing test module '/qsim/qsimcirq_tests/qsimcirq_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
qsimcirq_tests/qsimcirq_test.py:18: in <module>
    import cirq
/usr/local/lib/python3.7/dist-packages/cirq/__init__.py:21: in <module>
    from cirq import (
/usr/local/lib/python3.7/dist-packages/cirq/_compat.py:22: in <module>
    import pandas as pd
/usr/local/lib/python3.7/dist-packages/pandas/__init__.py:22: in <module>
    from pandas.compat.numpy import (
/usr/local/lib/python3.7/dist-packages/pandas/compat/numpy/__init__.py:21: in <module>
    f"this version of pandas is incompatible with numpy < {_min_numpy_ver}\n"
E   ImportError: this version of pandas is incompatible with numpy < 1.16.5
E   your numpy version is 1.16.2.
E   Please upgrade numpy to >= 1.16.5 to use this pandas version
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 0.78 seconds ============================
make: Leaving directory '/qsim'
make: *** [Makefile:40: run-py-tests] Error 1
```
This bears similarity to a recent Cirq issue: see [Cirq PR #3572](https://github.com/quantumlib/Cirq/pull/3572) and related issues. That issue was only partially traced before opting to turn down Python 3.6 support as an alternate fix.

--- a/pybind_interface/Dockerfile
+++ b/pybind_interface/Dockerfile
@@ -3,7 +3,9 @@ FROM qsim
 
 # Install additional requirements
 RUN apt-get install -y python3-dev python3-pybind11 python3-pytest python3-pip
-RUN pip3 install cirq
+
+# --force fixes the old numpy installation from pybind
+RUN pip3 install cirq --force
 
 # Copy relevant files
 COPY ./pybind_interface/ /qsim/pybind_interface/

--- a/pybind_interface/Dockerfile
+++ b/pybind_interface/Dockerfile
@@ -4,7 +4,8 @@ FROM qsim
 # Install additional requirements
 RUN apt-get install -y python3-dev python3-pybind11 python3-pytest python3-pip
 
-# --force fixes the old numpy installation from pybind
+# The --force flag is used mainly so that the old numpy installation from pybind
+# gets replaced with the one cirq requires
 RUN pip3 install cirq --force
 
 # Copy relevant files

